src/utils/generator.js

```diff
@@ -61,7 +73,23 @@ export type DerivedExpressionBuildNodeFunction = (
 
 export type GeneratorBuildNodeFunction = (Array<BabelNodeExpression>, SerializationContext) => BabelNodeStatement;
 
-export type GeneratorEntry = {
+export class GeneratorEntry {
+  constructor(generator: Generator) {
+    this.containingGenerator = generator;
+  }
+
+  containingGenerator: Generator;
```

Comment:
Let's try to avoid state if we can. I think we had issues in the past when GeneratorEntries were moved across Generators --- we should not do that anymore, but still...

Instead of maintaining this extra state, could the `visit` function get the containing generator as an argument?

Reply:
I will make an invariant for that then because these entries *cannot* be moved to another generator as they require respective effects to be applied which will only happen while serializing that particular generator.